Fixing GemmaScope SAEs listed in pretrained_saes.yaml

JoshEngels opened the pull request:
# Description

Some of the GemmaScope SAEs on SAE Lens don't really exist, and vice versa. I basically made this fix manually. This script identifies them (at some point would be good to commit a check like this):

```
# %%
from huggingface_hub import HfApi
from sae_lens.toolkit.pretrained_saes_directory import get_pretrained_saes_directory

for repo_id in ["google/gemma-scope-2b-pt-res", "google/gemma-scope-9b-pt-res", "google/gemma-scope-27b-pt-res"]:

    def get_repo_files(repo_id: str):
        """
        Get all file names from a Hugging Face repository
        
        Args:
            repo_id (str): Repository ID (e.g., 'bert-base-uncased' or 'username/repo-name')
        """
        api = HfApi()
        files = api.list_repo_files(repo_id)
        return files

    huggingface_files = get_repo_files(repo_id)

    sae_lens_files = get_pretrained_saes_directory()[repo_id.split("/")[-1]].saes_map

    huggingface_files = ["/".join(file.split("/")[:-1]) for file in huggingface_files if file.endswith(".npz")]
    huggingface_files = list(set(huggingface_files))


    # Find files in HuggingFace repo but not in sae_lens
    hf_only = set(huggingface_files) - set(sae_lens_files.values())

    # Find files in sae_lens but not in HuggingFace repo 
    sae_lens_only = set(sae_lens_files.values()) - set(huggingface_files)

    print("Files in HuggingFace repo but not in sae_lens:")
    for file in sorted(hf_only):
        print(f"{repo_id}  {file}")

    print("\nFiles in sae_lens but not in HuggingFace repo:")
    for file in sorted(sae_lens_only):
        print(f"{repo_id}  {file}")

# %%
` ``

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update



# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I have not rewritten tests relating to key interfaces which would affect backward compatibility

<!--
As you go through the checklist above, you can mark something as done by putting an x character in it

For example,
- [x] I have done this task
- [ ] I have not done this task
-->

### You have tested formatting, typing and unit tests (acceptance tests not currently in use)

- [ ] I have run `make check-ci` to check format and linting. (you can run `make format` to format code if needed.)

### Performance Check.

If you have implemented a training change, please indicate precisely how performance changes with respect to the following metrics:
- [ ] L0
- [ ] CE Loss
- [ ] MSE Loss
- [ ] Feature Dashboard Interpretability

Please links to wandb dashboards with a control and test group.

chanind commented:
I would have thought the docs build in CI should have caught this, but maybe that check has been scaled back or removed at some point (or I'm just misremembering). Definitely agree we should be checking for this stuff automatically

JoshEngels commented:
Makes sense, the only thing I'm a bit unsure about is that a couple of the things I deleted had neuronpedia links, and also the canonical 524k SAE didn't exist, so I changed it to the closest l0 to 100.

hijohnnylin commented:
thanks - these changes look good. some responses/comments:

- it looks like many of these are due to recent updates in the HF gemma scope repo: 
https://huggingface.co/google/gemma-scope-9b-pt-res/commit/f9b689815814972562d28082f9f7d65d7e01fdc8
https://huggingface.co/google/gemma-scope-2b-pt-res/commits/main/layer_12/width_131k
- agreed we should check this automatically (and/or periodically)
- longer term we need a better solution than keeping a huge yaml file in a repo - exposing easier methods to both list and load from HF directly is probably the way to go
- re: "deleted neuronpedia links" - these are fine since the neuronpedia references were only for an older set of SAEBench data (there are not actual dashboards generated for these, only evals) - and SAEBench is now mostly looking at gemma-2-2b

seems ok to merge from me. cc @curt-tigges

hijohnnylin commented:
added new issue https://github.com/jbloomAus/SAELens/issues/409
